futures-util/src/stream/fuse.rs

```diff
@@ -14,6 +15,7 @@ pub struct Fuse<S> {
     done: bool,
```

Comment:
Not new to this CL, but can you change this to use an `Option` (or create an issue to follow up and assign me)? That should allow for a more optimized representation, in addition to allowing the stream to drop before the `Fuse` is ready to be dropped.